data/lib/service/tournament/tournament_service.dart

```diff
@@ -54,6 +54,32 @@ class TournamentService {
     }
   }
 
+  Future<TournamentModel> getTournamentById(String id) async {
+    try {
+      final snapshot = await _tournamentCollection.doc(id).get();
+      var tournament = snapshot.data();
+      if (tournament != null) {
+        final teamIds = tournament.team_ids;
+        if (teamIds.isNotEmpty) {
+          final teams = await _teamService.getTeamsByIds(teamIds);
+          tournament = tournament.copyWith(teams: teams);
+        }
```

Comment:
I think we haven't used teams anywhere, so I don’t think we need to fetch teams here.